### PR TITLE
Add variant for Waveshare ESP32-S3-Touch-LCD-1.28

### PR DESCRIPTION
**Pull request by deisterhold**
## Description of Change
Adding variant for Waveshare ESP32-S3-Touch-LCD-1.28 board containing MCU, LCD, and IMU.

## Tests scenarios
Untested, pin definition is based on information in Waveshare wiki.

## Related links
Link to board wiki: https://www.waveshare.com/wiki/ESP32-S3-Touch-LCD-1.28

### CODE REVIEW

**Comment by me-no-dev** [comment]
It's missing the board definitions in `boards.txt`

**Comment by deisterhold** [comment]
Updated boards.txt

File: boards.txt [hunk 1]
```diff
@@ -34792,4 +34792,155 @@ Geekble_ESP32C3.menu.EraseFlash.none.upload.erase_cmd=
...
+waveshare_esp32s3_touch_lcd_128.upload.extra_flags=
+
+waveshare_esp32s3_touch_lcd_128.bootloader.tool=esptool_py
+waveshare_esp32s3_touch_lcd_128.bootloader.tool.default=esptool_py
+
+waveshare_esp32s3_touch_lcd_128.serial.disableDTR=true
+waveshare_esp32s3_touch_lcd_128.serial.disableRTS=true
+
+waveshare_esp32s3_touch_lcd_128.build.tarch=xtensa
+waveshare_esp32s3_touch_lcd_128.build.bootloader_addr=0x1000
```
Comment (Jason2866) [inline boards.txt:34818]: For any S3 the bootloader has to be at 0x0. Please change this wrong entry.

File: boards.txt [hunk 2]
```diff
@@ -34792,4 +34792,155 @@ Geekble_ESP32C3.menu.EraseFlash.none.upload.erase_cmd=
...
+waveshare_esp32s3_touch_lcd_128.build.flash_size=16MB
+waveshare_esp32s3_touch_lcd_128.build.flash_freq=80m
+waveshare_esp32s3_touch_lcd_128.build.flash_mode=dio
+waveshare_esp32s3_touch_lcd_128.build.boot=dio
+waveshare_esp32s3_touch_lcd_128.build.partitions=default
+
+waveshare_esp32s3_touch_lcd_128.menu.PSRAM.disabled=Disabled
+waveshare_esp32s3_touch_lcd_128.menu.PSRAM.disabled.build.defines=
+waveshare_esp32s3_touch_lcd_128.menu.PSRAM.enabled=Enabled
+waveshare_esp32s3_touch_lcd_128.menu.PSRAM.enabled.build.defines=-DBOARD_HAS_PSRAM -mfix-esp32-psram-cache-issue -mfix-esp32-psram-cache-strategy=memw
```
Comment (Jason2866) [inline boards.txt:34835]: remove this `-mfix-esp32-psram-cache-issue -mfix-esp32-psram-cache-strategy=memw`
It is not needed and wrong. Only early esp32 revisions with PSRAM does need this.

Comment (deisterhold) [reply boards.txt:34835]: Fixed.

File: boards.txt (hunk 1 above)
Comment (deisterhold) [reply boards.txt:34818]: Fixed.

File: boards.txt [hunk 3]
```diff
@@ -35828,4 +35828,155 @@ Geekble_ESP32C3.menu.EraseFlash.none.upload.erase_cmd=
...
+waveshare_esp32s3_touch_lcd_128.menu.FlashMode.dio.build.flash_mode=dio
+waveshare_esp32s3_touch_lcd_128.menu.FlashMode.dio.build.boot=dio
+
+waveshare_esp32s3_touch_lcd_128.menu.FlashFreq.80=80MHz
+waveshare_esp32s3_touch_lcd_128.menu.FlashFreq.80.build.flash_freq=80m
+waveshare_esp32s3_touch_lcd_128.menu.FlashFreq.40=40MHz
+waveshare_esp32s3_touch_lcd_128.menu.FlashFreq.40.build.flash_freq=40m
+
+waveshare_esp32s3_touch_lcd_128.menu.FlashSize.4M=4MB (32Mb)
+waveshare_esp32s3_touch_lcd_128.menu.FlashSize.4M.build.flash_size=4MB
```
Comment (P-R-O-C-H-Y) [inline boards.txt:35945]: This can be removed, as the board is made only with 16 MB of flash.
```suggestion
```

**Review by P-R-O-C-H-Y — COMMENTED** [review]
Looking good, just one small suggestion :)

File: boards.txt [hunk 4]
```diff
@@ -35828,4 +35828,155 @@ Geekble_ESP32C3.menu.EraseFlash.none.upload.erase_cmd=
...
+waveshare_esp32s3_touch_lcd_128.name=Waveshare ESP32S3 Touch LCD 128
+waveshare_esp32s3_touch_lcd_128.vid.0=0x1a86
+waveshare_esp32s3_touch_lcd_128.pid.0=0x55d3
+
+waveshare_esp32s3_touch_lcd_128.upload.tool=esptool_py
+waveshare_esp32s3_touch_lcd_128.upload.tool.default=esptool_py
+waveshare_esp32s3_touch_lcd_128.upload.tool.network=esp_ota
+waveshare_esp32s3_touch_lcd_128.upload.maximum_size=16777216
+waveshare_esp32s3_touch_lcd_128.upload.maximum_data_size=327680
+waveshare_esp32s3_touch_lcd_128.upload.wait_for_upload_port=true
```
Comment (P-R-O-C-H-Y) [inline boards.txt:35842]:
```suggestion
waveshare_esp32s3_touch_lcd_128.upload.wait_for_upload_port=false
```

File: boards.txt [hunk 5]
```diff
@@ -35828,4 +35828,155 @@ Geekble_ESP32C3.menu.EraseFlash.none.upload.erase_cmd=
...
+waveshare_esp32s3_touch_lcd_128.serial.disableRTS=true
+
+waveshare_esp32s3_touch_lcd_128.build.tarch=xtensa
+waveshare_esp32s3_touch_lcd_128.build.bootloader_addr=0x0
+waveshare_esp32s3_touch_lcd_128.build.mcu=esp32s3
+waveshare_esp32s3_touch_lcd_128.build.core=esp32
+waveshare_esp32s3_touch_lcd_128.build.target=esp32s3
+waveshare_esp32s3_touch_lcd_128.build.variant=waveshare_esp32s3_touch_lcd_128
+waveshare_esp32s3_touch_lcd_128.build.board=WAVESHARE_ESP32S3_TOUCH_LCD_128
+
```
Comment (P-R-O-C-H-Y) [inline boards.txt:35860]:
```suggestion

waveshare_esp32s3_touch_lcd_128.build.usb_mode=1
waveshare_esp32s3_touch_lcd_128.build.cdc_on_boot=1
waveshare_esp32s3_touch_lcd_128.build.msc_on_boot=0
waveshare_esp32s3_touch_lcd_128.build.dfu_on_boot=0
```

File: boards.txt [hunk 6]
```diff
@@ -35828,4 +35828,155 @@ Geekble_ESP32C3.menu.EraseFlash.none.upload.erase_cmd=
...
+waveshare_esp32s3_touch_lcd_128.build.f_cpu=240000000L
+waveshare_esp32s3_touch_lcd_128.build.flash_size=16MB
+waveshare_esp32s3_touch_lcd_128.build.flash_freq=80m
+waveshare_esp32s3_touch_lcd_128.build.flash_mode=dio
+waveshare_esp32s3_touch_lcd_128.build.boot=dio
+waveshare_esp32s3_touch_lcd_128.build.partitions=default
+
```
Comment (P-R-O-C-H-Y) [inline boards.txt:35867]:
```suggestion

waveshare_esp32s3_touch_lcd_128.menu.USBMode.hwcdc=Hardware CDC and JTAG
waveshare_esp32s3_touch_lcd_128.menu.USBMode.hwcdc.build.usb_mode=1
waveshare_esp32s3_touch_lcd_128.menu.USBMode.hwcdc.upload.use_1200bps_touch=false
waveshare_esp32s3_touch_lcd_128.menu.USBMode.hwcdc.upload.wait_for_upload_port=false
waveshare_esp32s3_touch_lcd_128.menu.USBMode.default=USB-OTG
waveshare_esp32s3_touch_lcd_128.menu.USBMode.default.build.usb_mode=0
waveshare_esp32s3_touch_lcd_128.menu.USBMode.default.upload.use_1200bps_touch=true
waveshare_esp32s3_touch_lcd_128.menu.USBMode.default.upload.wait_for_upload_port=true

waveshare_esp32s3_touch_lcd_128.menu.MSCOnBoot.default=Disabled
waveshare_esp32s3_touch_lcd_128.menu.MSCOnBoot.default.build.msc_on_boot=0
waveshare_esp32s3_touch_lcd_128.menu.MSCOnBoot.msc=Enabled (Requires USB-OTG Mode)
waveshare_esp32s3_touch_lcd_128.menu.MSCOnBoot.msc.build.msc_on_boot=1

waveshare_esp32s3_touch_lcd_128.menu.DFUOnBoot.default=Disabled
waveshare_esp32s3_touch_lcd_128.menu.DFUOnBoot.default.build.dfu_on_boot=0
waveshare_esp32s3_touch_lcd_128.menu.DFUOnBoot.dfu=Enabled (Requires USB-OTG Mode)
waveshare_esp32s3_touch_lcd_128.menu.DFUOnBoot.dfu.build.dfu_on_boot=1

```

**Review by P-R-O-C-H-Y — CHANGES_REQUESTED** [review]
CI Boards test is failing, due to missing USB stuff in boards.txt
I have added a suggestions what needs to be added/changed.

Have you tested if compilation works for you in the Arduino IDE?
Without those changes it fails. PTAL and after adding those changes test the board if you have one :) Thanks

File: boards.txt (hunk 6 above)
Comment (deisterhold) [reply boards.txt:35867]: Made all changes based on suggestions. Will test later this evening.

Comment (deisterhold) [reply boards.txt:35867]: Tested changes and it's compiling/running well.

File: boards.txt (hunk 5 above)
Comment (deisterhold) [reply boards.txt:35860]: Is this need if the board uses an external USB to UART chip?

Comment (P-R-O-C-H-Y) [reply boards.txt:35860]: Those are functions of USB are related to the USB-OTG mode, which can be used. So yes :)

Comment (P-R-O-C-H-Y) [reply boards.txt:35860]: So with the default settings we added:
```
waveshare_esp32s3_touch_lcd_128.build.usb_mode=1
waveshare_esp32s3_touch_lcd_128.build.cdc_on_boot=1
waveshare_esp32s3_touch_lcd_128.build.msc_on_boot=0
waveshare_esp32s3_touch_lcd_128.build.dfu_on_boot=0
```
The board is working fine right? You can see the output in the Serial monitor.

Comment (deisterhold) [reply boards.txt:35860]: I can see serial output if I use Serial1. The built-in USB connection is not broken out.

Comment (me-no-dev) [reply boards.txt:35860]: board is serial-only. Uses a crappy WCH chip for USB to Serial.

Comment (deisterhold) [reply boards.txt:35860]: Yep.

**Review by P-R-O-C-H-Y — APPROVED** [review]
Looking good now :) 
@deisterhold There are some conflicts in boards.txt file, please fix them so we can merge.

**Pull request merged**